cmd/gpu_fakedev/gpu_fakedev.go

```diff
@@ -291,6 +297,79 @@ func generateDriFiles(opts genOptions) {
 	log.Printf("Done, created %d dirs, %d devices and %d files.", opts.dirs, opts.devs, opts.files)
 }
 
+func makeXelinkSideCar(topology string, gpus, tiles int, connections string) {
+	if topology != fullyConnected {
+		log.Printf("XELINK: generate xelink sidecar label file, using (GPUs: %d, Tiles: %d)", gpus, tiles)
+		saveSideCarFile(connections)
+	} else {
+		log.Printf("XELINK: generate xelink sidecar label file, using (GPUs: %d, Tiles: %d, Topology: %s)", gpus, tiles, topology)
+		saveSideCarFile(buildConnectionList(gpus, tiles))
+	}
+}
+
+func buildConnectionList(gpus, tiles int) string {
+	var nodes = make([]string, 0)
+
+	for mm := 0; mm < gpus; mm++ {
+		for nn := 0; nn < tiles; nn++ {
+			nodes = append(nodes, fmt.Sprintf("%d.%d", mm, nn))
+		}
+	}
+
+	var links = make(map[string]struct{}, 0)
+
+	var smap = make([]string, 0)
+
+	for _, from := range nodes {
+		for _, to := range nodes {
+			// no self links, TODO ignore in-gpu xelinks
+			if to == from {
+				continue
+			}
+
+			link := fmt.Sprintf("%s-%s", to, from)
+
+			reverselink := fmt.Sprintf("%s-%s", from, to)
+			if _, exists := links[reverselink]; !exists {
+				links[link] = struct{}{}
+
+				smap = append(smap, link)
+			}
+		}
+	}
+
+	return fmt.Sprintf("%s", strings.Join(smap, "_"))
```

Comment:
```suggestion
	return strings.Join(smap, "_")
```
You don't need the Sprintf anymore.

Reply:
No that better without!